File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
def dfs(numbers,lists,path):
    if path == '0':
            return
    if len(path):
        lists.add(int(path))
    
    for i in range(len(numbers)):
        new_path = path[:]
        new_numbers = numbers[:]
        new_path += numbers[i]
        new_numbers.pop(i)
        dfs(new_numbers, lists, new_path)
    
def solution(numbers):
    answer = 0
    numbers = [str(n) for n in numbers]
    lists= set()
    dfs(numbers,lists,"")
    
    for num in lists:
        if num != 0 and num !=1:
            flag = False
            for j in range(2,num):  # 소수 판별
                if num % j == 0:
                    flag=True
                    break
            if flag == False:
                answer +=1
                
    return answer

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js

function solution(numbers) {
    var answer = 0;
    const set = new Set([]);
    
    dfs("", numbers, set);
    set.forEach((value)=> {
        if(isPrime(value)) answer++;
    });
    
    return answer;
}

function dfs(curr, numbers, set) {
    if (curr !== "") set.add(Number(curr));
    for (let i = 0; i<numbers.length; i++){
        dfs(curr+numbers[i],numbers.slice(0,i)+numbers.slice(i+1),set);
    }
}

function isPrime(number) {
    if (number === 0 ||number === 1) return false;
    for (let n = 2; n<number; n++){
        if (number%n ===0) return false;
    }
    return true;
}